File: frontend/src/factories/StocksFactory.js
import Stock from "@/domain/Stock.js";
import StockItemsFactory from "@/factories/StockItemsFactory.js";

export default class StocksFactory {
    static build() {
        return new StocksFactory(StockItemsFactory.build());
    }

    constructor(stockItemsFactory) {
        this._stockItemsFactory = stockItemsFactory;
    }

    createCollection(stocks) {
        return stocks.map(stock => this.create(stock));
    }

    create(stock) {
        const stockItems = this._stockItemsFactory.createCollection(stock.stockItems);
        return new Stock(
            stock.id,
            stock.name,
            stockItems
        );
    }
}

File: backend/src/test/java/com/erp/server/factories/StocksFactory.java
package com.erp.server.factories;

import infra.global.entities.StockEntity;
import org.instancio.Instancio;
import org.instancio.Select;

import java.util.ArrayList;

public class StocksFactory {

    public StockEntity createStock() {
        return Instancio.of(StockEntity.class)
                .set(Select.field("id"), null)
                .set(Select.field("deleted"), false)
                .set(Select.field("name"), "Test Stock")
                .set(Select.field("stockItems"), new ArrayList<>())
                .create();
    }
}
